bindings/python/scripts/dlite-validate

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+"""Validator for dlite datamodels.
+"""
+import argparse
+import json
+import re
+import yaml
+from pathlib import Path
+
+import dlite
+
+
+def parse(url, driver, options=None):
+    """Parse url and return an instance."""
+    match = re.match(r'^([a-zA-Z][a-zA-Z0-9+.-]*)://', url)
+    if match and match.groups[0].lower() != 'file':
+        if driver:
+            return dlite.Instance.from_location(driver, url, options=options)
+        else:
+            return dlite.Instance.from_url(url, options=options)
+    else:
+        path = url[7:] if url.lower().startswith('file://') else url
+        if driver:
+            return dlite.Instance_from_location(driver, path, options=options)
+        fileext = Path(path).suffix.lstrip('.').lower()
+
+        # For json and yaml, use the Python parser instead of dlite, since
+        # they have better error reporting for invalid serialisation
+        if fileext == 'json':
+            with open(path, 'rt') as f:
+                return dlite.Instance.from_dict(json.load(f))
+        elif fileext in ('yaml', 'yml'):
+            with open(path, 'rt') as f:
+                return dlite.Instance.from_dict(yaml.save_load(f))
+        return dlite.Instance.from_location(fileext, path, options=options)
+
+
+def main():
+    """Datamodel validator"""
+    parser = argparse.ArgumentParser(
+        description='DLite datamodel validator.',
+    )
+    parser.add_argument(
+        'url',
+        help='URL to storage to validate.  Relative file paths are allowed.'
+    )
+    parser.add_argument(
+        '--driver', '--plugin', '-d',
+        help='Storage plugin to use when loading the instance.  If `url` '
+        'is a file path, the default is to infer the driver from the file '
+        'extension.',
+    )
+    parser.add_argument(
+        '--options', '-o',
+        help='Options parsed on to the dlite storage plugin.'
```

Comment:
What kind of options are these?

Reply:
dlite.Instance.from_location() takes four arguments, driver, location, options and id. This flag make it possible to provide a value to the options argument. This is not needed for validating a datamodel stored in json or yaml files, but may be used to e.g. provide username/password for validating a datamodel stored in a postgresql database.

I updated the help text to: "Semicolon-separated list of options parsed on to the dlite storage plugin."
Should we include more of the above explanation?

Reply:
I think at least one example would be good. Otherwise it is very cryptic what kind of options these really are.

Reply:
Ok, da har jeg lagt til et eksempel.

La også til et par ekstra tests med forenklet syntax via et par brukerdefinerte cmake-funksjoner.